### ISSUE TITLE
Accessibility errors with Axe Dev Tools scanner

### ISSUE THREAD

**Issue by smlombardi** [ng-bootstrap/ng-bootstrap#4090]
### Bug description:
I am using browser-based accessibility scanning tools to determine our apps' accessibility.  I use mostly Axe dev tools from Deque systems.  For some reason it _always_ flags our ng-bootstrap tabs (ngbNav) elements as having missing and improper html structure.

I have double-checked our HTML. I have also forked your tabs example in stackblitz and if you check it with Axe or Accessibility Tools, you get the same errors:

![ng-bootstrap nav demo - basic-20210512_131748](https://user-images.githubusercontent.com/835148/118017570-c7f74e00-b324-11eb-9be9-98b273b1eabe.png)

<img width="1132" alt="Automated checks - Accessibility Insights for Web-20210512_131928" src="https://user-images.githubusercontent.com/835148/118017588-ce85c580-b324-11eb-949f-d308c7007be3.png">

### Link to minimally-working StackBlitz that reproduces the issue: 

https://stackblitz.com/edit/angular-dudcmr?file=src/index.html

### Versions of Angular, ng-bootstrap and Bootstrap:

These are the versions in your fork-able stackblitz

Angular: 11.0.9

ng-bootstrap: 9.1.1

Bootstrap: 4.5.3

**Comment by radupurdea** [comment]
I am experiencing the same issue.

According to [tab="role"](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/tab_role), there are 2 ways to link a tab to a tablist:
_Elements with the role tab must either be a child of an element with the tablist role, or have their id as part of the **aria-owns** property of a tablist._

Meaning you can do the following to fix this accessibility issue:

```
<ul ngbNav #nav="ngbNav" class="nav-tabs" aria-owns="tab1 tab2">
  <li ngbNavItem domId="tab1">
    <a ngbNavLink id="tab1">First</a>
    <ng-template ngbNavContent>First content</ng-template>
  </li>
  <li ngbNavItem domId="tab2">
    <a ngbNavLink id="tab2">Second</a>
    <ng-template ngbNavContent>Second content</ng-template>
  </li>
</ul>
<div [ngbNavOutlet]="nav"></div>
```

Here I've used `domId` to set the ngbNavLink anchor id. I added the ngbNavLink id only because ES Lint would complain about it at design time.

Hope it helps!

**Comment by clementcontet** [comment]
Hi @radupurdea doing this indeed fixes the issue with the elements with `tab` role not having a parent with `tablist` role.

But then we got another problem: Axe says that the `<li>` elements must have a parent with a `list` role.

I tried to force the `<ul>` to have both `list` and `tablist` roles with `roles=false role="tablist list"` but apparently an element cannot have more than one active role.

**Comment by stephanieleary** [comment]
I believe the underlying problem here is that `role="tab"` needs to be placed on the `<li>` element instead of the `<a>`. Moving this logic from the `NgbNavLink `directive to `NgbNavItem `should fix the issue.

**Comment by maxokorokov** [comment]
Closed by #4414 and #4488
